File: Network Log Simulator/tryRequest.py
import asyncio
import requests
import subprocess

async def fetch_data_stream(url):
    try:
        with requests.get(url, stream=True) as response:
            # Check if the response is successful
            if response.status_code == 200:
                # Iterate over the stream and print each line of data
                for line in response.iter_lines():
                    if line:
                        print(line.decode('utf-8'))  # Assuming data is UTF-8 encoded
            else:
                print(f"Failed to fetch data: {response.status_code}")
    except requests.RequestException as e:
        print(f"Error fetching data: {e}")

async def main():
    # URL of the data stream API
    url = 'http://localhost:5001/v1/logs/logs'  # Replace with your API endpoint

    # Fetch data stream asynchronously
    await fetch_data_stream(url)

# Run the asynchronous event loop
if __name__ == "__main__":
    # asyncio.run(main())
    # subprocess.run(["cd","/Users/vanilla/College work/Capstone/keystone/Network Log Simulator/"], shell=True)
    # result = subprocess.run(["cd ./Network\ Log\ Simulator/ && (npm run start &)"], shell=True, capture_output=True, text=True) #&&","ls","-l
    # print(f"Output: {result.stdout}, Error: {result.stderr}")
    subprocess.Popen(["(npm run start &)"], shell=True, cwd='Network Log Simulator')
    print("Triggered Network Logs Simulator, Use 'ps -ax | grep Server.js' to kill the npm server")

File: Network Log Simulator/tryRequest.js
var url = 'http://localhost:4000/v1/userlocation/userlocation?name=DBH%206th%20Floor'
fetch(url)
  .then(response => {
    const reader = response.body.getReader();
    const decoder = new TextDecoder();
    let accumulatedData = '';

    // Define a function to read the stream
    const readStream = () => {
      reader.read().then(({ done, value }) => {
        if (done) {
          console.log('Stream finished');
          // Parse the accumulated data as JSON
          // const jsonData = JSON.parse(accumulatedData);
          // console.log('Received data:', jsonData);
          return;
        }

        // Decode the chunk of data
        const chunk = decoder.decode(value, { stream: true });
        const parsed = JSON.parse(chunk);
        console.log(parsed);
        
        // Accumulate the chunk of data
        accumulatedData += chunk;

        // Continue reading the stream
        readStream();
      }).catch(error => {
        console.error('Error reading stream:', error);
      });
    };

    // Start reading the stream
    readStream();
  })
  .catch(error => {
    console.error('Error fetching data:', error);
  });